multibranch/src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java

```diff
@@ -0,0 +1,140 @@
+    @Rule public GitSampleRepoRule sampleRepo = new GitSampleRepoRule();
+
+    @SuppressWarnings("rawtypes")
+    @Test public void configRoundTripParameters() throws Exception {
+        StepConfigTester tester = new StepConfigTester(r);
+        assertEquals(Collections.emptyList(), tester.configRoundTrip(new JobPropertyStep(Collections.<JobProperty>emptyList())).getProperties());
+        List<JobProperty> properties = tester.configRoundTrip(new JobPropertyStep(Collections.<JobProperty>singletonList(new ParametersDefinitionProperty(new BooleanParameterDefinition("flag", true, null))))).getProperties();
+        assertEquals(1, properties.size());
+        assertEquals(ParametersDefinitionProperty.class, properties.get(0).getClass());
+        ParametersDefinitionProperty pdp = (ParametersDefinitionProperty) properties.get(0);
+        assertEquals(1, pdp.getParameterDefinitions().size());
+        assertEquals(BooleanParameterDefinition.class, pdp.getParameterDefinitions().get(0).getClass());
+        BooleanParameterDefinition bpd = (BooleanParameterDefinition) pdp.getParameterDefinitions().get(0);
+        assertEquals("flag", bpd.getName());
+        assertTrue(bpd.isDefaultValue());
+        // TODO JENKINS-29711 means it seems to omit the required () but we are not currently testing the Snippetizer output anyway
+    }
+
+    @SuppressWarnings("rawtypes")
+    @Test public void configRoundTripBuildDiscarder() throws Exception {
+        StepConfigTester tester = new StepConfigTester(r);
+        assertEquals(Collections.emptyList(), tester.configRoundTrip(new JobPropertyStep(Collections.<JobProperty>emptyList())).getProperties());
+        List<JobProperty> properties = tester.configRoundTrip(new JobPropertyStep(Collections.<JobProperty>singletonList(new BuildDiscarderProperty(new LogRotator(1, 2, -1, 3))))).getProperties();
+        assertEquals(1, properties.size());
+        assertEquals(BuildDiscarderProperty.class, properties.get(0).getClass());
+        BuildDiscarderProperty bdp = (BuildDiscarderProperty) properties.get(0);
+        BuildDiscarder strategy = bdp.getStrategy();
+        assertNotNull(strategy);
+        assertEquals(LogRotator.class, strategy.getClass());
+        LogRotator lr = (LogRotator) strategy;
+        assertEquals(1, lr.getDaysToKeep());
+        assertEquals(2, lr.getNumToKeep());
+        assertEquals(-1, lr.getArtifactDaysToKeep());
+        assertEquals(3, lr.getArtifactNumToKeep());
+    }
+
+    @Issue("JENKINS-30206")
+    @Test public void useParameter() throws Exception {
+        sampleRepo.init();
+        ScriptApproval.get().approveSignature("method groovy.lang.Binding hasVariable java.lang.String"); // TODO add to generic whitelist
+        sampleRepo.write("Jenkinsfile",
+                "properties([[$class: 'ParametersDefinitionProperty', parameterDefinitions: [[$class: 'StringParameterDefinition', name: 'myparam', defaultValue: 'default value']]]])\n" +
```

Comment:
I know it's what we have for now, but I have to say this is the longest way to define a `key=value` pair I have ever seen in my life.